extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -248,5 +251,212 @@ export class ProjectsController {
+
+			//Refresh project to show the added files
+			this.refreshProjectsTree();
+		}
+		catch (err) {
+			this.apiWrapper.showErrorMessage(utils.getErrorMessage(err));
+		}
+	}
+
+	private async getProjectName(dbName: string): Promise<string | undefined> {
+		let projName = await this.apiWrapper.showInputBox({
+			prompt: constants.newDatabaseProjectName,
+			value: `DatabaseProject${dbName}`
+		});
+
+		projName = projName?.trim();
+
+		return projName;
+	}
+
+	private async getExtractTarget(): Promise<ExtractTarget> {
+		let extractTarget: ExtractTarget;
+
+		const extractTargetOptions: QuickPickItem[] = [];
+
+		let keys: string[] = Object.keys(ExtractTarget).filter(k => typeof ExtractTarget[k as any] === 'number');
+
+		keys.forEach(targetOption => {
+			let pascalCaseTargetOption: string = utils.toPascalCase(targetOption);	// for better readability
+			extractTargetOptions.push({ label: pascalCaseTargetOption });
+		});
+
+		let input = await this.apiWrapper.showQuickPick(extractTargetOptions.slice(1), {		//Ignore the first option to create Dacpac
+			canPickMany: false,
+			placeHolder: constants.extractTargetInput
+		});
+		let extractTargetInput = input?.label;
+
+		if (extractTargetInput) {
+			let camelCaseInput: string = utils.toCamelCase(extractTargetInput);
+			extractTarget = ExtractTarget[camelCaseInput as keyof typeof ExtractTarget];
+		} else {
+			extractTarget = -1;
+		}
+
+		return extractTarget;
+	}
+
+	private async getFolderLocation(extractTarget: ExtractTarget): Promise<Uri | undefined> {
+		let selectionResult;
+		let projUri;
+
+		if (extractTarget !== ExtractTarget.file) {
+			selectionResult = await this.apiWrapper.showOpenDialog({
+				canSelectFiles: false,
+				canSelectFolders: true,
+				canSelectMany: false,
+				openLabel: constants.selectFileFolder,
+				defaultUri: this.apiWrapper.workspaceFolders() ? (this.apiWrapper.workspaceFolders() as WorkspaceFolder[])[0].uri : undefined
+			});
+			if (selectionResult) {
+				projUri = (selectionResult as Uri[])[0];
+			}
+		} else {
+			// Get filename
+			selectionResult = await this.apiWrapper.showSaveDialog(
+				{
+					defaultUri: this.apiWrapper.workspaceFolders() ? (this.apiWrapper.workspaceFolders() as WorkspaceFolder[])[0].uri : undefined,
+					saveLabel: constants.selectFileFolder,
+					filters: {
+						'SQL files': ['sql'],
+						'All files': ['*']
+					}
+				}
+			);
+			if (selectionResult) {
+				projUri = selectionResult as unknown as Uri;
+			}
+		}
+
+		return projUri;
+	}
+
+	private async isDirEmpty(newProjFolderUri: string): Promise<boolean> {
+		return (await fs.readdir(newProjFolderUri)).length === 0;
+	}
+
+	private async importApiCall(model: ImportDataModel): Promise<void> {
+		let ext = extensions.getExtension(mssql.extension.name);
+
+		if (ext === undefined) {
+			this.apiWrapper.showErrorMessage('VSCode extension undefined');
+			return;
+		}
+
+		const service = (ext.exports as mssql.IExtension).dacFx;
```

Comment:
It's preferred to call activate instead - that will return the mssql.IExension API and also has the benefit of guaranteeing that the extension is activated fully.